### DISCUSSION TITLE
Rke2 Ingress Controller won’t Issue dapr service

### DISCUSSION THREAD

**Discussion by Mjwagner2277** [rancher/rke2#4786] [Q&A]
I’m honestly unsure if this is a dapr issue or an rke Ingress Issue so I’m asking in both places. I applied the dapr annotations to the rke2-ingress-nginx-controller daemonset and the dapr container successfully injects but something is preventing the dapr operator from creating a Service of the dapr-ized controller pods. Any thoughts on if this is expected behavior?

**Comment by brandond** [comment]
> something is preventing the dapr operator from creating a Service of the dapr-ized controller pods.

I'm not familiar with dapr so you're going to have to be more specific about what's going on. How did you deploy it? What annotations have you added, and what is the expected result? Is there an error in the operator's controller logs? Have you turned up the log level to see if it is even trying to do anything?

**Comment by Mjwagner2277** [comment]
https://alphatango086.medium.com/nginx-ingress-controller-with-dapr-a69e03e98209

This is the instructions for installing dapr given to me by the dapr community(they are in process of developing official documentation). 

The annotations are the default ones for enabling injection of the dapr sidecar. The expected result is that the dapr sidecar container get injected(this occurs) and then the dapr-operator pod creates a kubernetes service associated with the pod that was just containerized(this does not occur).
 
The operator logs do not show an attempt or a failure to create the Kubernetes service as it would with other pods and the log level is debug by default. 

I was trying to figure out via the helm charts if something would prevent a pod from creating services within namespaces but I don’t see anything indicative of that. 

 I am pretty new to Kubernetes(in my first 3 months as a developer in general) and I am just not understanding why the dapr-operator would simply fail to attempt to create a service. Again, I’m not sure if this is a dapr operator issue or some kind of  rbac issue within rke2-ingress-nginx-controller
